metaflow/plugins/aws/batch/batch_client.py

```diff
@@ -14,42 +15,118 @@
 
 from metaflow.exception import MetaflowException
 from metaflow.metaflow_config import AWS_SANDBOX_ENABLED
+from metaflow.metaflow_config import BATCH_DESCRIBE_JOBS_THROTTLE_DELTA as DELTA
+from metaflow.metaflow_config import BATCH_DESCRIBE_JOBS_THROTTLE_TRIES as TRIES
+
+
+class Throttle(object):
+    def __init__(self, delta_in_secs=DELTA, num_tries=TRIES):
+        self._now = time.time()
+        self.delta_in_secs = int(delta_in_secs)  # Cast if from env variable
+        self.num_tries = int(num_tries)  # Cast if from env variable
+        self._reset()
+
+    def _reset(self):
+        self._tries_left = self.num_tries
+
+    def __call__(self, func):
+        def wrapped(*args, **kwargs):
+            # Don't poll incessantly
+            wait = self.delta_in_secs - (time.time() - self._now)
+            if wait > 0:
+                time.sleep(wait)
```

Comment:
Hmm I'm not sure I follow this logic, looks like this would wait even before the very first call?

Reply:
I've switched to the google-style exponential backoff, as suggested, so this is now resolved